File: Frontend/Mousiq/src/services/contact.service.ts
import {apiClient} from './api.service';
import type {Contact, ContactRequest} from '../types/contact.types';
import type {MessageResponse} from '../types/api.types';

export const contactService = {
    getUserContacts: async (): Promise<Contact[]> => {
        const response = await apiClient.get<Contact[]>('/api/contacts');
        return response.data;
    },

    addContact: async (request: ContactRequest): Promise<MessageResponse> => {
        const response = await apiClient.post<MessageResponse>('/api/contacts', request);
        return response.data;
    },

    editContact: async (id: string, request: ContactRequest): Promise<MessageResponse> => {
        const response = await apiClient.put<MessageResponse>(`/api/contacts/${id}`, request);
        return response.data;
    },

    deleteContact: async (id: string): Promise<void> => {
        await apiClient.delete(`/api/contacts/${id}`);
    },
};

export default contactService;

File: BackendMonolit/Mousiq/src/main/java/org/example/mousiq/contact/service/ContactService.java
package org.example.mousiq.contact.service;

import lombok.RequiredArgsConstructor;
import org.example.mousiq.contact.model.dto.ContactDTO;
import org.example.mousiq.contact.model.entity.Contact;
import org.example.mousiq.user.model.entity.User;
import org.example.mousiq.common.exception.ForbiddenException;
import org.example.mousiq.common.exception.NotFoundException;
import org.example.mousiq.contact.model.request.ContactRequest;
import org.example.mousiq.contact.repository.ContactRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class ContactService {
    private final ContactRepository contactRepository;

    @Transactional
    public ContactDTO addContact(ContactRequest request, User user) {
        Contact contact = Contact.builder()
                .email(request.email())
                .phoneNumber(request.phoneNumber())
                .user(user).build();

        return new ContactDTO(saveContact(contact));
    }

    public Contact saveContact(Contact contact) {
        return contactRepository.save(contact);
    }

    public Contact findContactById(UUID id) {
        return contactRepository.findById(id)
                .orElseThrow(() -> new NotFoundException("Contact with id: " + id + " do not exist"));
    }

    @Transactional
    public void deleteUserContact(UUID id, User user) {
        Contact contact = findContactById(id);

        if (!contact.getUser().equals(user)) {
            throw new ForbiddenException("User do not contains this contact");
        }

        deleteContact(contact);
    }

    public void deleteContact(Contact contact) {
        contactRepository.delete(contact);
    }

    @Transactional
    public ContactDTO editUserContact(UUID id, ContactRequest request, User user) {
        Contact contact = findContactById(id);

        if (!contact.getUser().equals(user)) {
            throw new ForbiddenException("User do not contains this contact");
        }

        contact.setEmail(request.email());
        contact.setPhoneNumber(request.phoneNumber());

        return new ContactDTO(saveContact(contact));
    }

    public List<ContactDTO> getUserContacts(User user) {
        return contactRepository.findContactsByUser(user)
                .stream()
                .map(ContactDTO::new)
                .collect(Collectors.toList());
    }
}